amr-wind/equation_systems/icns/source_terms/BodyForce.cpp

```diff
@@ -0,0 +1,60 @@
+#include "amr-wind/equation_systems/icns/source_terms/BodyForce.H"
+#include "amr-wind/CFDSim.H"
+#include "amr-wind/utilities/trig_ops.H"
+
+#include "AMReX_ParmParse.H"
+#include "AMReX_Gpu.H"
+
+namespace amr_wind {
+namespace pde {
+namespace icns {
+
+/** Body Force
+ */
+BodyForce::BodyForce(const CFDSim& sim)
+    : m_time(sim.time()), m_density(sim.repo().get_field("density"))
+{
+
+    {
+        // Read the geostrophic wind speed vector (in m/s)
+        amrex::ParmParse pp("BodyForce");
+        pp.query("type", m_type);
+        pp.queryarr("magnitude", m_body_force);
+        if (m_type == "Oscillatory" || m_type == "oscillatory") {
```

Comment:
I would recommend doing this instead:

```C++
pp.query("type", "m_type");
m_type = amrex::toLower(m_type);

if (m_type == "oscillatory") pp.query("angular_frequency", m_omega);
``` 

so that user can also do `OsCiLlaToRy` and still have it work